src/utilities/utils.js

```diff
@@ -7,23 +7,50 @@ export const cleanData = data => {
       colors: currentPolish.product_colors
     }
   })
-  console.log(freshData)
 
-//reduce through the array
-//inside reduce, filter through the current array's polish brands
-//return only the brands with duplicates in array
-//then, map over this array of duplicates to create a new array of all colors
-//.flat that bb
-//create a new array without duplicates using Set
-//reset the currentPolishes colors to be this new Set
-//push the currentPolish into the accumulator
-//return the reduce
+  const findUnrepeated = array => [...new Set(array)]
 
+  const justBrandNames = freshData.map(element => {
+    return element.brand
+  })
 
-return freshData
+  const uniqueBrands = findUnrepeated(justBrandNames)
 
+  const organizedData = uniqueBrands.map(name => {
+    const newData = {}
+    newData.colors = []
+    freshData.forEach(polObj => {
+      if (polObj.brand === name) {
+        newData.brand = name;
+        newData.id = polObj.id;
+        newData.image = polObj.image;
+        polObj.colors.forEach(color => {
+          newData.colors.push(color)
+        })
+      }
+    })
+    const justColorNames = newData.colors.map(color => {
+      return color.colour_name
+    })
+    const uniqueColors = findUnrepeated(justColorNames)
+    const cleanColors = uniqueColors.map(color => {
+      const newColor = {}
+      newData.colors.forEach(origCol => {
+        if (origCol.colour_name === color) {
+          newColor.colorway = color
+          newColor.hue = origCol.hex_value
+        }
+      })
+      return newColor
+    })
+    newData.colors = cleanColors
```

Comment:
I think 46 and 47 wouldn't be needed if you did the refactor in my previous comment?